src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectState.cs

```diff
@@ -440,6 +440,13 @@ public List<string> GetImportDocumentTargetPaths(HostDocument hostDocument)
             foreach (var importItem in importItems)
             {
                 var itemTargetPath = importItem.FilePath.Replace('/', '\\').TrimStart('\\');
+
+                if (FilePathComparer.Instance.Equals(itemTargetPath, hostDocument.TargetPath))
```

Comment:
This seems like a workaround rather than a proper fix. Is the issue that `hostDocument.TargetPath` isn't normalized correctly? Could we fix that?

Reply:
At first I thought it was more of a workaround too but in an effort to not re-remember the motivations for the above `importItem.FilePath.Replace('/', '\\')` I made the assumption that it was by design (I can't recall why we normalize there and didn't want to break anything old). Aka imagine a `HostDocument` on Linux/Mac:

```JSON
{
    "TargetPath": "_Imports.razor",
    "Filepath": "/path/project/_Imports.razor"
}
```

Trying to get a list of potential impactful import paths equates to:
- `/path/_Imports.razor`
- `/_Imports.razor`

When we normalize the second result we come up with `_Imports.razor` which matches the original `TargetPath`.

This becomes impactful even in Windows LanguageServer scenarios because our LanguageServer gets fed OS agnostic file paths from LanguageServer clients (VSCode) i.e. `/c:/path/project/file.razor` resulting in the `/_Imports.razor` case being normalized down into the original target path.

I could take the approach of re-purposing: `var itemTargetPath = importItem.FilePath.Replace('/', '\\').TrimStart('\\');` however I worry i'll break existing VS4Mac or VS scenarios.

Let me know your thoughts or if you feel differently given the above info.